pubsub/google/cloud/pubsub/connection.py

```diff
@@ -419,7 +424,11 @@ def subscription_pull(self, subscription_path, return_immediately=False,
             'maxMessages': max_messages,
         }
         response = conn.api_request(method='POST', path=path, data=data)
-        return response.get('receivedMessages', ())
+        messages = response.get('receivedMessages', ())
+        for key, _ in enumerate(messages):
+            messages[key]['message']['data'] = base64.b64decode(
+                messages[key]['message']['data'])
```

Comment:
Having to deal with the extra `'message'` key is a bit less than desired but I wouldn't say it makes that code weird. You could instead say

```python
def _json_message_base64(messages, transform, key=None):
    for message in messages:
        if key is not None:
            message = message[key]
        if 'data' in message:
            message['data'] = transform(message['data'])
```